enterprise/coderd/provisionerkeys.go

```diff
@@ -54,6 +58,19 @@ func (api *API) postProvisionerKey(rw http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	if slices.Contains(codersdk.ReservedProvisionerKeyNames(), req.Name) {
```

Comment:
Might want to make this:
```suggestion
if slices.ContainsFunc(codersdk.ReservedProvisionerKeyNames(), func(s string) bool {
		return strings.EqualFold(req.Name, s)
	}) {
```

To prevent different casings. Would be confusing is all.